### PR TITLE
Fix size of varchenko_matrix

### LINKED ISSUE #36490: A bug with the function varchenko_matrix()
### Steps To Reproduce

H.<x , y> = HyperplaneArrangements(QQ)
g = y | x-y | x+y+1
g.varchenko_matrix()
[Varchenko-Matrix.pdf](https://github.com/sagemath/sage/files/13053331/Varchenko-Matrix.pdf)

### Expected Behavior

We should obtain a 7x7-matrix

### Actual Behavior

We obtain a 3x3-matrix

### Additional Information

_No response_

### Environment

```markdown
- **OS**: macOS Sonoma 14.0
- **Sage Version**: SageMath-10-1
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by DaveWitteMorris**
Fixes #36490

The code for the function `varchenko_matrix` erroneously assumes that the number n of regions is equal to the number k of hyperplanes, and therefore returns a k-by-k matrix, instead of an n-by-n matrix. This PR changes k to n where appropriate.

It also adds a doctest and fixes two existing doctests.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.

### CODE REVIEW

File: src/sage/geometry/hyperplane_arrangement/arrangement.py [hunk 1]
```diff
@@ -3157,21 +3157,35 @@ def varchenko_matrix(self, names='h'):
...
         h = R.gens()
         region = self.regions()
-        v = identity_matrix(R, k, k)
-        for i in range(k):
-            for j in range(i+1, k):
+        n = len(region)
+        v = identity_matrix(R, n, n)
+        for i in range(n):
+            for j in range(i+1, n):
                 t = prod(h[p] for p in range(k) if
```
Comment (kwankyu) [inline src/sage/geometry/hyperplane_arrangement/arrangement.py:3189]: `i + 1`?

**Comment by kwankyu** [comment]
Otherwise, lgtm.

File: src/sage/geometry/hyperplane_arrangement/arrangement.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/geometry/hyperplane_arrangement/arrangement.py:3189]: done

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by DaveWitteMorris** [comment]
Thanks, Kwankyu and Frédéric!

**Pull request merged**